[CI] SharedClusterSnapshotRestoreIT.testThrottling failure

**Build scan**: https://gradle-enterprise.elastic.co/s/pui6fsuhb34yu

**Repro line**: 
```
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT.testThrottling" \
  -Dtests.seed=8B1E1125079B7194 \
  -Dtests.security.manager=true \
  -Dtests.locale=de-CH \
  -Dtests.timezone=Pacific/Marquesas \
  -Druntime.java=11
```
**Reproduces locally?**: No

**Applicable branches**: master, 7.x

**Failure history**:
[July 21](https://gradle-enterprise.elastic.co/s/uylc7jktwmsjw)
[July 20](https://gradle-enterprise.elastic.co/s/t63pn4bak2t2e)

**Failure excerpt**:
```
java.lang.AssertionError: 
Expected: <0L>
     but: was <1112317L>
	at __randomizedtesting.SeedInfo.seed([8B1E1125079B7194:DCCEEE1E65F7EFE8]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT.testThrottling(SharedClusterSnapshotRestoreIT.java:1853)
```

--- a/server/src/internalClusterTest/java/org/elasticsearch/snapshots/SharedClusterSnapshotRestoreIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/snapshots/SharedClusterSnapshotRestoreIT.java
@@ -1825,10 +1825,9 @@ public class SharedClusterSnapshotRestoreIT extends AbstractSnapshotIntegTestCas
         cluster().wipeIndices("test-idx");
 
         logger.info("--> restore index");
-        if (throttleRestoreViaRecoverySettings) {
-            client.admin().cluster().prepareUpdateSettings().setTransientSettings(Settings.builder()
-                .put(INDICES_RECOVERY_MAX_BYTES_PER_SEC_SETTING.getKey(), "10k").build()).get();
-        }
+        client.admin().cluster().prepareUpdateSettings().setTransientSettings(Settings.builder()
+                .put(INDICES_RECOVERY_MAX_BYTES_PER_SEC_SETTING.getKey(),
+                        throttleRestoreViaRecoverySettings ? "10k" : "0").build()).get();
         RestoreSnapshotResponse restoreSnapshotResponse = client.admin().cluster().prepareRestoreSnapshot("test-repo", "test-snap")
             .setWaitForCompletion(true).execute().actionGet();
         assertThat(restoreSnapshotResponse.getRestoreInfo().totalShards(), greaterThan(0));

